src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/PackageFeeds/SourceRepositoryExtensions.cs

```diff
@@ -88,6 +88,10 @@ public static Task<SearchResult<IPackageSearchMetadata>> SearchAsync(this Source
 
             using (var sourceCacheContext = new SourceCacheContext())
             {
+                // Update http source cache context MaxAge so that it can always go online to fetch
```

Comment:
Could you change the method API, make the caller can control whether it goes online to fetch? It's ok for UI, but I think it should can be controlled by caller, since this is a method in SourceRepository

Reply:
This code path is only called through Install/ Update/ Consolidate tabs which is where it should always fetch from Server. Do you know if this is also called from any other place than those UI tab then let me know.